src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -145,81 +147,32 @@ public unsafe BitArray(bool[] values)
             // (true for any non-zero values, false for 0) - any values between 2-255 will be interpreted as false.
             // Instead, We compare with zeroes (== false) then negate the result to ensure compatibility.
 
-            if (Avx2.IsSupported)
+            ref byte value = ref Unsafe.As<bool, byte>(ref MemoryMarshal.GetArrayDataReference<bool>(values));
+
+            if (Vector256.IsHardwareAccelerated)
             {
-                // JIT does not support code hoisting for SIMD yet
-                Vector256<byte> zero = Vector256<byte>.Zero;
-                fixed (bool* ptr = values)
+                for (; (i + Vector256ByteCount) <= (uint)values.Length; i += Vector256ByteCount)
```

Comment:
```suggestion
                for (; i <= (uint)values.Length - Vector256ByteCount; i += Vector256ByteCount)
```
? So the repeated addition to the loop variable can be avoided, as the JIT will keep "length - constant" in a register.

Same below in the Vector128 code path.

Reply:
right. The JIT will effectively generate this as `[base + index]` on x86/x64 and will, after we do a PR, generate a special form of `LDR` on ARM64